ding_dibs.email_reports.inc

```diff
@@ -0,0 +1,251 @@
+ *
+ * @return
+ *   Returns tabular representation of values results.
+ */
+function ding_dibs_email_reports() {
+  $time = REQUEST_TIME - 10800;
+  $report = db_select('dibs_transactions', 'd')
+    ->fields('d', array(
+      'payment_order_id',
+      'payment_transaction_id',
+      'payment_time',
+      'customer_uid',
+      'order_info_short',
+      'order_info_long',
+      'payment_price',
+      'payment_status',
+      'provider_payment_status',
+      'dibs_capture_status',
+    ))
+    ->condition('payment_time', date('Y-m-d H:i:s', $time), '>')
+    ->condition('d.api_module', 'ding_dibs')
+    ->condition('d.api_delta', 1)
+    ->orderBy('payment_order_id', 'DESC')
+    ->execute()
+    ->fetchAll();
+  $max_count = count($report);
+  if (!$max_count) {
+    return '<p>' . t('No transactions found.') . '</p>';
+  }
+  for ($count = 0; $count < $max_count; $count++) {
+    //Clipping of output orders with the status of "1/1/1" and "0/0/0"
+    if (!($report[$count]->payment_status == $report[$count]->provider_payment_status && $report[$count]->payment_status == $report[$count]->dibs_capture_status && $report[$count]->provider_payment_status == $report[$count]->dibs_capture_status)) {
+      $report_valid[] = $report[$count];
+    }
+  }
+  $header = array(
+    array('data' => t('Order ID')),
+    array('data' => t('Transaction ID')),
+    array('data' => t('Date')),
+    array('data' => t('User')),
+    array('data' => t('Text')),
+    array('data' => t('Amount')),
+    array('data' => t('Cleared')),
+    array('data' => t('Paid')),
+    array('data' => t('Captured')),
+  );
+  $rows = array();
+  foreach ($report_valid as $record) {
+    $rows[] = _ding_dibs_report_generation_values($record);
+  }
+
+  $output = theme('table', array(
+    'header' => $header,
+    'rows' => $rows,
+    'attributes' => array('id' => 'ding_dibs_email_reports'),
+  ));
+  return $output;
+}
+
+/**
+ * Callback that builds page reports run cron.
+ *
+ * @return
+ *   Returns tabular representation of values results.
+ */
+function ding_dibs_cron_report() {
+  $report = db_select('ding_dibs_report_cron', 'd')
+    ->fields('d', array('id', 'time_start', 'status', 'text'))
+    ->orderBy('id', 'DESC')
+    ->execute()
+    ->fetchAll();
+  $max_count = count($report);
+  if (!$max_count) {
+    return drupal_set_message(t('No cron report.'), 'warning');
+  }
+  drupal_add_css(drupal_get_path('module', 'ding_dibs') . '/css/ding_dibs.css');
+  $header = array(array('data' => t('Message'), 'class' => 'payment'));
+  $rows = array();
+  foreach ($report as $record) {
+    if (empty($record->text)) {
+      $day = format_date($record->time_start, 'custom', 'd', date_default_timezone_get());
+      $hour = format_date($record->time_start, 'custom', 'H:i', date_default_timezone_get());
+      $month = format_date($record->time_start, 'custom', 'F', date_default_timezone_get());
+      $record->text = $day . 'th ' . $month . ' ' . $hour . ': Job failed.';
```

Comment:
Outputting individual date elements here is unneeded. Same for adding "th".

This could be reduced to something like:

```php
$record->text = t('%date: Job failed', array('%date' => format_date($record->time_start, 'custom' , 'd F H:i'));
```